### DISCUSSION TITLE
Virtual Property: em.create() respects required properties

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#3061] [Q&A] [answered]
Hello.
I've recently updated from v4 to v5.
And I've seen this part of documentation:

> em.create() will now require you to pass all non-optional properties. Some properties might be defined as required for TS, but we have a default value for them (either runtime, or database one) - for such we can use OptionalProps symbol to specify which properties should be considered as optional.

And I've got a question about it. How should I deal with virtual properties?

E.g:
I've got this status virtual property and my create method is missing that property.
I don't know how to make it optional. I cannot put a question mark after the JS getter. So how should I proceed?
```
  @Property({ persist: false })
  get status(): EnumStatus {
      return EnumStatus.OK;
    }
```

Thank you

**Comment by B4nan** [comment, marked as answer]
I will need to document this somewhere, I believe I wrote about it but can't find it. You can use the `OptionalProps` symbol. See the entities in ORM tests for examples:

https://github.com/mikro-orm/mikro-orm/blob/master/tests/entities-sql/Author2.ts#L16

**Reply by kennethsato** [reply]
Oh, I'm sorry.
I thought the "OptionalProps symbol" you've mentioned was the question mark.
It's fine now.
Thanks @B4nan 

```
export default class Insercao {

  [OptionalProps]?: 'status' | 'created_at' | 'statusmapa';

 @Property({ persist: false })
  get status(): EnumStatus {
      return EnumStatus.OK;
    }
}
```

**Comment by kennethsato** [comment]
Hello!

Still about v4 to v5 upgrade, I've got the following error:

>  ValidationError: Value for Emissora.nome is required, 'undefined' found entity: Emissora { id: 1 }

I have many creates that used only the FK ID to create a relationship, for example:
```
      const mapa = this.ormRepository.create({
        emissora: 1
      });
```

I've tried to use OptionalProps to overcome that, but it didn't work.
Must I load my related entity before every insert/create?

**Reply by B4nan** [reply]
That is runtime validation, not ts error. Your property is not marked as nullable, so you need to provide a value for it.

**Reply by kennethsato** [reply]
I need to mark as nullable even though I have a value (FK ID) for it?

**Reply by B4nan** [reply]
What value? You call em.create() with an entity, it has a mandatory property called name, but you dont provide any value for it.
